File: nr1-main/validators/userValidator.js
// validators/userValidator.js
const Joi = require('joi');

const signup = Joi.object({
  email: Joi.string().email().required(),
  password: Joi.string().min(6).max(128).required(),
});

const login = Joi.object({
  email: Joi.string().email().required(),
  password: Joi.string().min(6).max(128).required(),
});

module.exports = { signup, login };

File: nr1-main/app/validators/user_validator.py
"""
Pydantic models for user request validation.
(Deprecated: Prefer using app/schemas.py for all request/response models.)
"""

from pydantic import BaseModel, Field
from typing import Optional

class UserSignupRequest(BaseModel):
    """Request model for user signup."""
    username: str = Field(..., min_length=3, max_length=32)
    password: str = Field(..., min_length=6)
    email: str = Field(..., description="User email")

class UserLoginRequest(BaseModel):
    """Request model for user login."""
    username: str = Field(...)
    password: str = Field(...)
